receipt_upload/receipt_upload/line_items/reconstructor.py

```
@@ -58,6 +59,178 @@ def _amount(t: str) -> float | None:
+        if (w.line_id, w.word_id) in pos
+    }
+    if not by_key:
+        return [], []
+
+    def cy(k: Tuple[int, int]) -> float:
+        return pos[k][1]
+
+    grand_keys = [
+        k
+        for k, labs in label_objs.items()
+        if any(l.label == "GRAND_TOTAL" for l in labs) and k in by_key
```

Ruff (0.15.17)

[error] Ambiguous variable name: `l`

(E741)